grease/src/Grease/Setup.hs

```diff
@@ -128,11 +132,49 @@ freshPtrBv ::
   Selector ext argTys ts regTy ->
   ValueName (Mem.LLVMPointerType w) ->
   NatRepr w ->
-  Setup sym ext argTys (Mem.LLVMPtr sym w)
+  Setup sym ext argTys w' (Mem.LLVMPtr sym w)
 freshPtrBv sym sel nm w =
-  annotatePtrBv sym sel
+  annotatePtrBv sym sel 
     =<< liftIO (W4.freshConstant sym (safeSymbol nm) (W4.BaseBVRepr w))
 
+
+
+
+
+
+setupPtrMem ::
+  forall sym bak ext tag w argTys ts regTy.
+  ( C.IsSymBackend sym bak
+  , C.IsSyntaxExtension ext
+  , Mem.HasPtrWidth w
+  , Mem.HasLLVMAnn sym
+  , ?memOpts :: Mem.MemOptions
+  , Cursor.CursorExt ext ~ PtrCursor.Dereference ext w
+  , Cursor.Last (regTy ': ts) ~ Mem.LLVMPointerType w
+  ) =>
+  GreaseLogAction ->
+  bak ->
+  Mem.DataLayout ->
+  ValueName (Mem.LLVMPointerType w) ->
+  Selector ext argTys ts regTy ->
+  PtrTarget w tag ->
+  Setup sym ext argTys w (C.RegValue sym (Mem.LLVMPointerType w), PtrTarget w (C.RegValue' sym))
+setupPtrMem la bak layout nm sel tgt@(PtrTarget _ bid) = 
+  let r = setupPtr la bak layout nm sel tgt in 
+  case bid of 
+    Just bid' -> do 
+      resmap <- use setupRes
```

Comment:
Please use `camelCase` for local variable names
```suggestion
      resMap <- use setupRes
```
https://galoisinc.github.io/grease/dev/style.html#naming